apps/src/Sounds.js

```diff
@@ -328,6 +372,67 @@ Sounds.prototype.playURL = function(url, playbackOptions) {
   }
 };
 
+/**
+ * @param {Promise} promise the promise that will resolve to the ArrayBuffer bytes to be played
+ * @param {object} playbackOptions config for the playing of the sound.
+ * @param {object} cacheParams optional parameters for caching the result of promise
+ */
+Sounds.prototype.addPromiseToSpeechQueue = function(
+  promise,
+  playbackOptions,
+  cacheParams = undefined
+) {
+  this.speechQueue.push({promise, playbackOptions, cacheParams});
+  this.checkSpeechQueue();
+};
+
+Sounds.prototype.checkSpeechQueue = async function() {
+  if (!this.speechPlaying && this.speechQueue.length > 0) {
+    this.speechPlaying = true;
+    let nextSpeech = this.speechQueue.shift();
+    let bytes = await nextSpeech.promise;
+    if (bytes === undefined) {
+      if (
+        this.textBytesByLanguage[nextSpeech.cacheParams.language][
```

Comment:
nit: we access `this.textBytesByLanguage[nextSpeech.cacheParams.language][nextSpeech.cacheParams.text]` a few times, so let's move this into a variable for readability